framework/filters/AccessRule.php

```diff
@@ -55,9 +56,38 @@ class AccessRule extends Component
      * In this case, [[User::can()]] will be called to check access.
      *
      * If this property is not set or empty, it means this rule applies to all roles.
+     * @see $roleParams
      */
     public $roles;
     /**
+     * @var array|Closure parameters to pass to the [[User::can()]] function for evaluating
+     * user permissions in [[$roles]].
+     *
+     * If this is an array, it will be passed directly to [[User::can()]].
+     * You may also specify a closure that returns an array. This can be used to
+     * evaluate the array values only if they are needed.
+     * This can be used for example like this:
+     *
+     * ```php
+     * 'rules' => [
+     *     [
+     *         'allow' => true,
+     *         'actions' => ['update'],
+     *         'roles' => ['updatePost'],
+     *         'roleParams' => function($rule) {
+     *             return ['postId' => Yii::$app->request->get('id')];
```

Comment:
This example does not need a closure. Because the `rules` array is built in a method of the controller class, `Yii:: $app->request` should already be populated, so one could set the `roleParams` array directly. I think it would be nice to have an example for each case, with closure and without.